catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/validation/impl/validator/CountryCodeValidator.java

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.validation.impl.validator;
+
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.Set;
+
+import org.apache.commons.lang.builder.EqualsBuilder;
+import org.apache.commons.lang.builder.HashCodeBuilder;
+import com.google.common.base.Preconditions;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.validation.AttributeValidator;
+import ddf.catalog.validation.impl.report.AttributeValidationReportImpl;
+import ddf.catalog.validation.impl.violation.ValidationViolationImpl;
+import ddf.catalog.validation.report.AttributeValidationReport;
+import ddf.catalog.validation.violation.ValidationViolation;
+
+/**
+ * Validates an attribute's value(s) against the ISO_3166-1 Alpha3 country codes.
+ * <p>
+ * Is capable of validating {@link String}s.
+ */
+public class CountryCodeValidator implements AttributeValidator {
+    private Set<String> countryCodesMap;
+
+    //TODO: Should there be an option to ignore case?
+    public CountryCodeValidator() {
+        setupISO3CountryCodes();
+    }
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     * Validates only the values of {@code attribute} that are {@link String}s.
+     */
+    @Override
+    public Optional<AttributeValidationReport> validate(Attribute attribute) {
+        Preconditions.checkArgument(attribute != null, "The attribute cannot be null.");
+
+        final String name = attribute.getName();
+        for (final Serializable value : attribute.getValues()) {
```

Comment:
Cool. Then you don't want that map operation, definitely.

You still might consider extracting a method like `AttributeValidationReport newMethod(String name, String countryCode)` to make the code cleaner, but otherwise it's good with me.